src/routes/Profile/Competences/GroupBySource.tsx

```diff
@@ -1,52 +1,129 @@
-              <div className={groupByHeaderClasses}>{t(`my-competences.by-${sourceType}`)}</div>
-              <div className="flex flex-wrap gap-3">
-                {osaamiset.map((val) => {
-                  const label = val.osaaminen.nimi[locale] ?? val.osaaminen.uri;
-                  const title = val.osaaminen.kuvaus[locale];
-                  return (
-                    val.lahde.tyyppi === sourceType &&
-                    isOsaaminenVisible(sourceType, val.lahde.id) && (
-                      <Tag
-                        label={label}
-                        title={title}
-                        key={val.id}
-                        variant="presentation"
-                        sourceType={OSAAMINEN_COLOR_MAP[val.lahde.tyyppi]}
-                      />
-                    )
-                  );
-                })}
-              </div>
+      <div className="mb-10">
+        {COMPETENCE_TYPES.map((competence) => {
+          return (
+            <React.Fragment key={competence}>
+              <div className={groupByHeaderClasses}>{t(`my-competences.by-${competence}`)}</div>
+              {Array.isArray(filters[competence]) && filters[competence].some((filter) => filter.checked) ? (
+                <div>
+                  <div className="flex flex-wrap gap-4">
+                    {osaamiset.map((val) => {
+                      const label = val.osaaminen.nimi[locale] ?? val.osaaminen.uri;
+                      const title = val.osaaminen.kuvaus[locale];
+                      return (
+                        val.lahde.tyyppi === competence &&
+                        isOsaaminenVisible(competence, val.lahde.id) && (
+                          <Tag
+                            label={label}
+                            title={title}
+                            key={val.id}
+                            variant="presentation"
+                            sourceType={OSAAMINEN_COLOR_MAP[val.lahde.tyyppi]}
+                          />
+                        )
+                      );
+                    })}
+                  </div>
+                  <Link
+                    to={competenceLinks(competence)}
+                    key={competence}
+                    type="button"
+                    className="text-button-md hover:underline px-5 py-3 text-accent mt-3"
+                  >
+                    <div className="flex flex-row justify-start">
+                      <div className="flex items-center gap-3">
+                        {t('profile.competences.edit')}
+                        <ArrowIcon />
+                      </div>
+                    </div>
+                  </Link>
+                </div>
+              ) : (
+                <Link
+                  to={competenceLinks(competence)}
+                  key={competence}
+                  type="button"
+                  className="text-button-md hover:underline px-5 py-3 text-accent"
```

Comment:
```suggestion
                  className="text-button-md hover:underline mt-3 text-accent"
```

Reply:
Removed paddings and used margin where needed